lib/bolt/transport/orch/connection.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+module Bolt
+  module Transport
+    class Orch < Base
+      class Connection
+        attr_reader :logger, :key
+
+        CONTEXT_KEYS = Set.new(%i[plan_name description params]).freeze
+
+        def self.get_key(opts)
+          [
+            opts['service-url'],
+            opts['task-environment'],
+            opts['token-file']
+          ].join('-')
+        end
+
+        def initialize(opts, plan_context, logger)
+          @logger = logger
+          @key = self.class.get_key(opts)
+          client_keys = %w[service-url token-file cacert]
+          client_opts = client_keys.each_with_object({}) do |k, acc|
+            acc[k] = opts[k] if opts.include?(k)
+          end
+          logger.debug("Creating orchestrator client for #{client_opts}")
+
+          @client = OrchestratorClient.new(client_opts, true)
+          @plan_job = start_plan(plan_context)
+
+          @environment = opts["task-environment"]
+        end
+
+        def start_plan(plan_context)
+          if plan_context
```

Comment:
Should this throw an error if there's no plan context? Or does it just return nil?

Reply:
I'm not sure how it would get called with no plan context and I don't know anywhere else we validate internal arguments like that.